src/blueprint/operators/call.rs

```diff
@@ -0,0 +1,340 @@
+use std::collections::hash_map::Iter;
+
+use crate::blueprint::*;
+use crate::config;
+use crate::config::group_by::GroupBy;
+use crate::config::{Config, Field, GraphQLOperationType};
+use crate::graphql;
+use crate::grpc;
+use crate::http;
+use crate::lambda::DataLoaderId;
+use crate::lambda::{Expression, IO};
+use crate::mustache::{Mustache, Segment};
+use crate::try_fold::TryFold;
+use crate::valid::Valid;
+
+fn find_value<'a>(args: &'a Iter<'a, String, String>, key: &'a String) -> Option<&'a String> {
+    args.clone()
+        .find_map(|(k, value)| if k == key { Some(value) } else { None })
+}
+
+pub fn update_call(
+    operation_type: &GraphQLOperationType,
+) -> TryFold<'_, (&Config, &Field, &config::Type, &str), FieldDefinition, String> {
+    TryFold::<(&Config, &Field, &config::Type, &str), FieldDefinition, String>::new(
+        move |(config, field, _, _), b_field| {
+            let Some(call) = &field.call else {
+                return Valid::succeed(b_field);
+            };
+
+            compile_call(field, config, call, operation_type)
+                .and_then(|resolver| Valid::succeed(b_field.resolver(Some(resolver))))
+        },
+    )
+}
+
+struct Http {
+    pub req_template: http::RequestTemplate,
+    pub group_by: Option<GroupBy>,
+    pub dl_id: Option<DataLoaderId>,
+}
+
+struct GraphQLEndpoint {
+    pub req_template: graphql::RequestTemplate,
+    pub field_name: String,
+    pub batch: bool,
+    pub dl_id: Option<DataLoaderId>,
+}
+
+struct Grpc {
+    pub req_template: grpc::RequestTemplate,
+    pub group_by: Option<GroupBy>,
+    pub dl_id: Option<DataLoaderId>,
+}
+
+impl TryFrom<Expression> for Http {
+    type Error = String;
+
+    fn try_from(expr: Expression) -> Result<Self, Self::Error> {
+        match expr {
+            Expression::IO(IO::Http { req_template, group_by, dl_id }) => {
+                Ok(Http { req_template, group_by, dl_id })
+            }
+            _ => Err("not an http expression".to_string()),
+        }
+    }
+}
+
+impl TryFrom<Expression> for GraphQLEndpoint {
+    type Error = String;
+
+    fn try_from(expr: Expression) -> Result<Self, Self::Error> {
+        match expr {
+            Expression::IO(IO::GraphQLEndpoint { req_template, field_name, batch, dl_id }) => {
+                Ok(GraphQLEndpoint { req_template, field_name, batch, dl_id })
+            }
+            _ => Err("not a graphql expression".to_string()),
+        }
+    }
+}
+
+impl TryFrom<Expression> for Grpc {
+    type Error = String;
+
+    fn try_from(expr: Expression) -> Result<Self, Self::Error> {
+        match expr {
+            Expression::IO(IO::Grpc { req_template, group_by, dl_id }) => {
+                Ok(Grpc { req_template, group_by, dl_id })
+            }
+            _ => Err("not a grpc expression".to_string()),
+        }
+    }
+}
```

Comment:
This was done to avoid unreachable code. I can delete it and use `match` letting an unreachable `_ => Valid::succeed` or `_ => panic!`, let me know which is better, guys.